### DISCUSSION TITLE
on: check_suite never gets called

### DISCUSSION THREAD

**Discussion by dpryan79** [community/community#26236] [Actions]
<p>It seems that <code>on: check_suite</code> is currently never being triggered. I have a workflow (<a href="https://github.com/dpryan79/libBigWig/blob/master/.github/workflows/artifacts.yml" rel="nofollow noopener">https://github.com/dpryan79/libBigWig/blob/master/.github/workflows/artifacts.yml</a> ) that is never actually getting called by this event, even after all CI checks are completed. Or is there something that needs to be changed to enable this?</p>

**Comment by weide-zhou** [comment]
<p>Hi,</p>
<p>Many thanks for your feedback!  </p>
<p>Events raised from the Actions app do not trigger workflows.  This restriction is currently in place in order to prevent a circular workflow execution.  </p>
<p>Here is a similar <a href="https://github.community/t/can-you-use-checks-api-actions-to-trigger-new-workflows/16349">ticket</a> with more details, please kindly check, thanks.</p>

**Comment by dpryan79** [comment]
<p>In this case I’m not creating the original check_run or check_suite with github actions, that should be coming from TravisCI.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Sorry for late response <a class="mention" href="/dpryan79">@dpryan79</a> !</p>
<p>Only GitHub Apps can create check suites and check runs. Travis CI created commit status but did not create any check runs for your commits. Eg:</p>
<p>Commit: <a href="https://github.com/dpryan79/libBigWig/commit/4eb1d152c1c178f14eeb9412c0587cc3bab6fef5" rel="nofollow noopener">dpryan79/libBigWig@4eb1d15</a></p>
<p>Commit status: <a href="https://api.github.com/repos/dpryan79/libBigWig/commits/4eb1d152c1c178f14eeb9412c0587cc3bab6fef5/statuses" rel="nofollow noopener">https://api.github.com/repos/dpryan79/libBigWig/commits/4eb1d152c1c178f14eeb9412c0587cc3bab6fef5/statuses</a></p>
<p>No check run : curl -v -H “Accept: application/vnd.github.antiope-preview+json” <a href="https://api.github.com/repos/dpryan79/libBigWig/commits/4eb1d152c1c178f14eeb9412c0587cc3bab6fef5/check-runs" rel="nofollow noopener">https://api.github.com/repos/dpryan79/libBigWig/commits/4eb1d152c1c178f14eeb9412c0587cc3bab6fef5/check-runs</a></p>
<p>Hence Travis CI will no trigger any check suite/check run.</p>
<p>As an alternative, you can use ‘<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#status-event-status" rel="nofollow noopener">commit status event</a>’ instead.</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181096175-55f2f6e5-3546-4c7e-9854-8f93566625af.png" alt="statusTravis.png" data-base62-sha1="r8h9jQrApiJaYWhOEWbWDKrv2zG" width="561" height="139"></p>

**Comment by dpryan79** [comment]
<p>Ahh, that explains things, thanks!</p>

**Comment by hardboiled** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> Am I reading this wrong, or does the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#check-suite-event-check_suite" rel="nofollow noopener">official github documentation</a> list “check_suite” as a valid action to listen against? It seems there are many tickets (like the one below) logged against similar issues. I’m confused why the documentation isn’t updated with this information clarifying that it is not supported.</p>
<p><a href="https://github.community/t/can-you-use-checks-api-actions-to-trigger-new-workflows/16349">https://github.community/t5/GitHub-Actions/Can-you-use-Checks-API-Actions-to-trigger-new-workflows/td-p/32704</a></p>

**Comment by hardboiled** [comment]
<p>To clarify, I am trying to do something like this within a local branch:</p>
<pre><code>name: 'Listen to check_suite'
on:
  check_suite:
    types: [rerequested, completed]
jobs:
  job-that-listens-to-check-suite:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: 'attempt step'
         uses: ./.github/actions/my-local-action
</code></pre>
<p> When I listen to a different event (e.g. pull_request) the action fires properly, but the check_suite event never fires, even though it is triggered by a manual “git push” from myself (which I would think would be considered as not sourced by the “github actions” app). Is it expected behavior for the check_suite event not to be received in this situation by my github workflow?</p>
<p>Are there any situations where “check_suite” would trigger a workflow or is <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#check-suite-event-check_suite" rel="nofollow noopener">the documentation</a> wrong and it’s not actually supported?</p>
